### PR TITLE
Refactored Profile Report

### PR DESCRIPTION
**Pull request by rockwotj**
Refactor the profile reporter to be an object, this generally cleans things up (no need to pass options around everywhere!) and allows for easy addition of features such as [turning off collapsing of paths](https://groups.google.com//forum/#!topic/firebase-talk/u37GhguiId4) without plumbing the options everywhere.

### CODE REVIEW

File: lib/profileReport.js [hunk 1]
```diff
@@ -92,86 +143,66 @@ function collectBandwidth(bytes, path, direction) {
...
+  var path = ProfileReport.pathString(data.path);
+  this.state.opCount++;
   switch (data.name) {
   case 'concurrent-connect':
     break;
   case 'concurrent-disconnect':
     break;
   case 'realtime-read':
-    collect(data, path, data.bytes, state.readSpeed, state.outband);
+    this.collect(data, path, data.bytes, this.state.readSpeed, this.state.outband);
```
Comment (mbleigh) [inline lib/profileReport.js]: If the profile report is now a class, do you need to pass readSpeed and outband? Or can you just call `this.state.readSpeed` and `this.state.outband` from the `collect` function directly? Or have two methods: `collectRead` and `collectWrite`?

Comment (rockwotj) [inline lib/profileReport.js]: It was just to keep only a single `collect` function, I'm okay with `collectRead` and `collectWrite`, do you like that more?

**Comment by rockwotj** [comment]
Yupp! Just a refactor for future changes. For example, if I wanted to add an option to turn off collapsing paths, I can just add the option in the constructor and reference it in `collapsePaths` once instead of needing to plumb that option everywhere.

File: lib/profileReport.js (hunk 1 above)
Comment (mbleigh) [inline lib/profileReport.js]: Yeah I think it's a bit cleaner.

Comment (rockwotj) [inline lib/profileReport.js]: SG, I'll update that tonight

**Pull request merged**